### ISSUE TITLE
Use of AWS roles vs. credentials

### ISSUE THREAD

**Issue by cloudgurus** [MoriTanosuke/glacieruploader#48]
Hi,
Thank you for the great software. 
Would it be possible to use AWS role for EC2 (http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials) instead of providing hard coded credentials in aws.properties file?

**Comment by MoriTanosuke** [comment]
Hey, thanks for the praise. ☺ Will take a look at the docs you provided. I don't use Roles myself, so I can't tell if and how to integrate them into the code yet.

Will update this issue when I get an idea.

**Comment by MoriTanosuke** [comment]
This might be of interest as well https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html

**Comment by cloudgurus** [comment]
I agree. I'm not a Java guy, but looks like BasicSessionCredentials object needs to be used: http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/?com/amazonaws/auth/BasicSessionCredentials.html
